src/librustdoc/json/conversions.rs

```diff
@@ -232,41 +238,52 @@ fn from_clean_item(item: clean::Item, tcx: TyCtxt<'_>) -> ItemEnum {
-        TyMethodItem(m) => ItemEnum::Method(from_function_method(m, false, header.unwrap(), tcx)),
-        ImplItem(i) => ItemEnum::Impl(i.into_tcx(tcx)),
-        StaticItem(s) => ItemEnum::Static(s.into_tcx(tcx)),
-        ForeignStaticItem(s) => ItemEnum::Static(s.into_tcx(tcx)),
+        ImportItem(i) => ItemEnum::Import(i.into_tcx(tcx, cache)),
+        StructItem(s) => ItemEnum::Struct(s.into_tcx(tcx, cache)),
+        UnionItem(u) => ItemEnum::Union(u.into_tcx(tcx, cache)),
+        StructFieldItem(f) => ItemEnum::StructField(f.into_tcx(tcx, cache)),
+        EnumItem(e) => ItemEnum::Enum(e.into_tcx(tcx, cache)),
+        VariantItem(v) => ItemEnum::Variant(v.into_tcx(tcx, cache)),
+        FunctionItem(f) => ItemEnum::Function(from_function(f, header.unwrap(), tcx, cache)),
+        ForeignFunctionItem(f) => ItemEnum::Function(from_function(f, header.unwrap(), tcx, cache)),
+        TraitItem(t) => ItemEnum::Trait(t.into_tcx(tcx, cache)),
+        TraitAliasItem(t) => ItemEnum::TraitAlias(t.into_tcx(tcx, cache)),
+        MethodItem(m, _) => {
+            ItemEnum::Method(from_function_method(m, true, header.unwrap(), tcx, cache))
+        }
+        TyMethodItem(m) => {
+            ItemEnum::Method(from_function_method(m, false, header.unwrap(), tcx, cache))
+        }
+        ImplItem(i) => ItemEnum::Impl(i.into_tcx(tcx, cache)),
+        StaticItem(s) => ItemEnum::Static(s.into_tcx(tcx, cache)),
+        ForeignStaticItem(s) => ItemEnum::Static(s.into_tcx(tcx, cache)),
         ForeignTypeItem => ItemEnum::ForeignType,
-        TypedefItem(t) => ItemEnum::Typedef(t.into_tcx(tcx)),
-        OpaqueTyItem(t) => ItemEnum::OpaqueTy(t.into_tcx(tcx)),
-        ConstantItem(c) => ItemEnum::Constant(c.into_tcx(tcx)),
+        TypedefItem(t) => ItemEnum::Typedef(t.into_tcx(tcx, cache)),
+        OpaqueTyItem(t) => ItemEnum::OpaqueTy(t.into_tcx(tcx, cache)),
+        ConstantItem(c) => ItemEnum::Constant(c.into_tcx(tcx, cache)),
         MacroItem(m) => ItemEnum::Macro(m.source),
-        ProcMacroItem(m) => ItemEnum::ProcMacro(m.into_tcx(tcx)),
+        ProcMacroItem(m) => ItemEnum::ProcMacro(m.into_tcx(tcx, cache)),
         PrimitiveItem(p) => ItemEnum::PrimitiveType(p.as_sym().to_string()),
-        TyAssocConstItem(ty) => ItemEnum::AssocConst { type_: ty.into_tcx(tcx), default: None },
-        AssocConstItem(ty, default) => {
-            ItemEnum::AssocConst { type_: ty.into_tcx(tcx), default: Some(default.expr(tcx)) }
-        }
+        TyAssocConstItem(ty) => ItemEnum::AssocConst {
+            type_: ty.into_tcx(tcx, cache),
+            default: None,
+            default_value: None,
+        },
+        AssocConstItem(ty, default) => ItemEnum::AssocConst {
+            type_: ty.into_tcx(tcx, cache),
+            default: Some(default.expr(tcx)),
+            // FIXME: Should we remove depth and length limits for the JSON backend?
```

Comment:
If the depth/lenght is limited, that should be documented in `rustdoc-json-types`